apps/site/assets/ts/schedule/components/line-diagram/VehicleIcons.tsx

```diff
@@ -5,26 +5,25 @@ import { RouteStop, LineDiagramVehicle } from "../__schedule";
 import CrowdingPill from "./CrowdingPill";
 import { TooltipWrapper, vehicleArrowIcon } from "../../../helpers/icon";
 import { StopCoord, CoordState, CIRC_RADIUS } from "./graphics/graphic-helpers";
-import { RouteType, HeadsignWithCrowding, Prediction } from "../../../__v3api";
 import {
   vehicleRealtimeStatusText,
   vehicleName
 } from "../../../models/vehicle";
-import { isACommuterRailRoute } from "../../../models/route";
-import { hasPredictionTime } from "../../../models/prediction";
+import { MapMarker } from "../../../leaflet/components/__mapdata";
 
 interface VehicleIconsProps {
   stop: RouteStop;
   vehicles: LineDiagramVehicle[] | null;
-  headsigns: HeadsignWithCrowding[];
+  vehicleMarkers: MapMarker[];
 }
 
-const tooltipText = (
-  routeType: RouteType | null,
-  stopName: string | null,
-  vehicle: LineDiagramVehicle,
-  prediction: Prediction | null
+// fallback to get the tooltip if vehicle channel data is not available
```

Comment:
This is a nice thought but I think we should proceed assuming the channel data is available! Please use the info from `vehicleMarkers` in place of the prior `vehicles` :)